### PR TITLE
Walk code tidy

### PR DESCRIPTION
**Pull request by Tom0Brien**
This PR does not change any functionality of the walk but cleans up some things:

- Naming conventions 
- How walk parameters are configured
- Use of `Vec3` and `Iso3` in class to help readability

### CODE REVIEW

File: module/input/SensorFilter/src/SensorFilter.hpp [hunk 1]
```diff
@@ -107,11 +107,11 @@ namespace module::input {
...
         tinyrobotics::Model<double, n_servos> nugus_model;
 
-        /// @brief Current support phase of the robot
-        WalkState::SupportPhase current_support_phase = WalkState::SupportPhase::LEFT;
+        /// @brief Current walk phase of the robot
+        WalkState::Phase current_walk_phase = WalkState::Phase::LEFT;
 
-        /// @brief Transform from anchor {a} to world {w} space
-        Eigen::Isometry3d Hwa = Eigen::Isometry3d::Identity();
+        /// @brief Transform from planted foot {a} to world {w} space
```
Comment (ysims) [inline module/input/SensorFilter/src/SensorFilter.hpp:113]:
```suggestion
        /// @brief Transform from planted foot {p} to world {w} space
```

File: shared/utility/skill/WalkGenerator.hpp [hunk 2]
```diff
@@ -242,64 +252,30 @@ namespace utility::skill {
...
-        /// @brief Torso offset at half step period from the planted foot, at time = step_apex_ratio * step_period
-        Eigen::Matrix<Scalar, 3, 1> torso_sway_offset = Eigen::Matrix<Scalar, 3, 1>::Zero();
-
-        /// @brief Ratio of the step_period where the torso should be at its maximum sway, between [0 1].
-        Scalar torso_sway_ratio = 0.0;
-
-        /// @brief Ratio of where to position the torso relative to the next step position [x,y] (in meters)
-        Eigen::Matrix<Scalar, 3, 1> torso_final_position_ratio = Eigen::Matrix<Scalar, 3, 1>::Zero();
+        /// @brief Walk engine parameters.
+        WalkParameters p;
```
Comment (ysims) [inline shared/utility/skill/WalkGenerator.hpp:256]: maybe `params` or similar? Single letter vars can get annoying when searching :laughing:

Comment (TrentHouliston) [reply shared/utility/skill/WalkGenerator.hpp:256]: The rule I use for single letter params is that you have to be able to see their entire usage from start to end within a single code screen

Comment (Tom0Brien) [reply shared/utility/skill/WalkGenerator.hpp:256]: WalkGenerator.hpp is the only file it's used in, this parameter struct is used a lot on the same lines which makes the code harder to read imo which is why I wanted to have it as a single letter

**Review by ysims — APPROVED** [review]

**Pull request merged**